src/tz_macos.rs

```diff
@@ -1,22 +1,47 @@
-use core_foundation_sys::base::{CFRelease, CFTypeRef};
-use core_foundation_sys::string::{kCFStringEncodingUTF8, CFStringGetCStringPtr};
+use core_foundation_sys::base::{Boolean, CFRange, CFRelease, CFTypeRef};
+use core_foundation_sys::string::{kCFStringEncodingUTF8, CFStringGetBytes, CFStringGetLength};
 use core_foundation_sys::timezone::{CFTimeZoneCopySystem, CFTimeZoneGetName};
 
 pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
-    unsafe {
-        Dropping::new(CFTimeZoneCopySystem())
-            .and_then(|tz| Dropping::new(CFTimeZoneGetName(tz.0)))
-            .and_then(|name| {
-                let name = CFStringGetCStringPtr(name.0, kCFStringEncodingUTF8);
-                if name.is_null() {
-                    None
-                } else {
-                    Some(name)
-                }
-            })
-            .and_then(|name| std::ffi::CStr::from_ptr(name).to_str().ok())
-            .map(|name| name.to_owned())
-            .ok_or(crate::GetTimezoneError::OsError)
+    unsafe { get_timezone().ok_or(crate::GetTimezoneError::OsError) }
+}
+
+#[inline]
+unsafe fn get_timezone() -> Option<String> {
+    // The longest name in the IANA time zone database is 25 ASCII characters long.
+    const MAX_LEN: usize = 32;
+
+    // Get system time zone, and its name.
+    let tz = Dropping::new(CFTimeZoneCopySystem())?;
+    let name = Dropping::new(CFTimeZoneGetName(tz.0))?;
```

Comment:
this is UB since `Dropping` will free an un-owned pointer.

See: https://developer.apple.com/documentation/corefoundation/1543519-cftimezonegetname

A string containing the geopolitical region name that identifies tz. Ownership follows the [The Get Rule](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-SW1).

> ## The Get Rule
>
> If you receive an object from any Core Foundation function other than a creation or copy function—such as a Get function—you do not own it and cannot be certain of the object’s life span. If you want to ensure that such an object is not disposed of while you are using it, you must claim ownership (with the CFRetain function). You are then responsible for relinquishing ownership when you have finished with it.

Reply:
Ah, thanks, fixed in the next commit. I guess that this could cause a leak, when the reference counter wraps around because it was decremented once too often.